react/src/pages/analysis/Analysis.tsx

```diff
@@ -231,38 +345,69 @@ export default function Analysis() {
                         pageSize: 10,
                         filtering: true,
                         search: false,
-                        padding: 'dense'
+                        padding: 'dense',
+                        selection: true
                     }}
                     actions={[
                         {
-                            icon: () => <Visibility className={classes.visibilityIcon} />,
+                            icon: Visibility,
                             tooltip: 'Analysis details',
+                            position: 'row',
                             onClick: (event, rowData) => {
-                                setActiveRow((rowData as AnalysisRow));
+                                // We can only view details of one row at a time
+                                setActiveRows([rowData as AnalysisRow]);
                                 setDetail(true);
                                 history.push(`/analysis/${(rowData as AnalysisRow).analysis_id}`);
-                            },
+                            }
                         },
-                        rowData => ({
+                        {
                             icon: Cancel,
                             tooltip: 'Cancel analysis',
-                            onClick: () => {
-                                setActiveRow(rowData)
-                                setCancel(true)
+                            onClick: (event, rowData) => {
+                                setActiveRows(rowData as AnalysisRow[]);
+                                setCancel(true);
                             },
-                            disabled: rowData.state !== PipelineStatus.RUNNING && rowData.state !== PipelineStatus.PENDING,
```

Comment:
I need to revise this feature since the Cancel action is on the toolbar now when rows are selected. Right now the Cancel Analysis Dialog will tell you which analyses will actually be cancelled, and which ones will be skipped.

Reply:
I looked into this problem in MaterialTable to make the Cancel button greyed-out when the selected rows are incompatible (wrong state), and the work-around involves using an arrow function to get the rowData: https://github.com/mbrn/material-table/issues/609#issuecomment-525228694
However, this cannot be used for selection-based actions with multiple rows: https://github.com/mbrn/material-table/issues/2515
So it looks as though my hands are tied, and that having the dialog tell the user which rows are affected is the only reasonable solution.

Reply:
Hmm, that's rather unfortunate. I guess that's how it will have to be now then. In CancelAnalysisDialog, you can't cancel pending analyses so that should be updated to match then.